crates/turborepo-lib/src/shim.rs

```diff
@@ -317,8 +317,14 @@ impl LocalTurboState {
     // - `pnpm install`
     // - `npm install --install-strategy=linked`
     fn generate_linked_path(root_path: &AbsoluteSystemPath) -> Option<AbsoluteSystemPathBuf> {
-        let canonical_path =
-            fs_canonicalize(root_path.join_components(&["node_modules", "turbo", ".."])).ok()?;
+        let canonical_path = fs_canonicalize(
+            root_path
+                .as_path()
+                .join("node_modules")
+                .join("turbo")
+                .join(".."),
+        )
+        .ok()?;
```

Comment:
So now it uses `as_path` to expose the `Utf8Path` operations directly.

Reply:
If the two behave differently, that's probably something we ought to fix.